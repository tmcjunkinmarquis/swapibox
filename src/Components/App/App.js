import React, { Component } from 'react';
import FilmCrawl from '../FilmCrawl/FilmCrawl';
import { CardContainer } from '../CardContainer/CardContainer';
import { 
  getHomeWorld, 
  getSpecies, 
  getPlanetDetails, 
  getVehicleDetails
} from '../../api/helper';
import { fetchApi } from '../../api/api';
import './App.css';
import PropTypes from 'prop-types';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      cardType: '',
      film: [],
      people: [],
      vehicles: [],
      planets: [],
      cleanedCardData: [],
      initialFetchedObj: []
    };
  }

  handleClick = (event) => {
    this.setState({ 
      cardType: event.target.value
    }, () => { 
      this.initialFetchCall(); 
    });
  }

  initialFetchCall = async () => {
    const categoryData = await fetchApi(this.state.cardType);

    this.checkCardType(categoryData);
  }

  getpeopleData = async (categoryData) => {
    const homeWorldObj = await getHomeWorld(categoryData);
    const speciesObj = await getSpecies(categoryData);
    const cleanedCardData = this.combineObj(homeWorldObj, speciesObj);

    this.setState({ people: cleanedCardData });
  }

  getVehicleData = async (categoryData) => {
    const vehicleObj = await getVehicleDetails(categoryData);

    this.setState({ vehicles: vehicleObj });
  }

  getPlanetData = async (categoryData) => {
    const planetObj = await getPlanetDetails(categoryData);

    this.setState({planets: planetObj});
  }

  checkCardType = async (categoryData) => {
    if (this.state.cardType === 'people') {
      this.getpeopleData(categoryData);
    } else if (this.state.cardType === 'planets') {
      this.getPlanetData(categoryData);
    } else if (this.state.cardType === 'vehicles') {
      this.getVehicleData(categoryData);
    }

    localStorage.setItem([this.props.cardType], 
      JSON.stringify(this.state[this.props.cardType]));
  }

  combineObj = (homeWorldObj, speciesObj) => {
    let combo;
    const combinedObj = homeWorldObj.reduce((acc, item, index) => {
      combo = Object.assign({}, speciesObj[index]);
      acc.push({ ...item, ...combo });
      return acc;
    }, []);
    return combinedObj;
  }


  // checkForLocalStorage = () => {
  //   !localStorage[this.state.cardType] ? this.initialFetchCall() : this.getFromLocalStorage()
  // }

  // getFromLocalStorage = () => {
  //   const retrieveData = localStorage.getItem((this.state.cardType)
  //   const parsedData = JSON.parse(retrieveData)
  // }

  render() {
    return (
      <div className="App">
        <div crawl="crawl">
          <FilmCrawl />
        </div>
        <div buttons="buttons">
          <button 
            onClick={this.handleClick} 
            value="people" 
            className="button">People
          </button>
          <button 
            onClick={this.handleClick} 
            value="planets" 
            className="button">Planets
          </button>
          <button 
            onClick={this.handleClick} 
            value="vehicles" 
            className="button">Vehicles
          </button>
          <button 
            className="button">Favorites
          </button>
        </div>
        <div>
          {this.state.cardType && 
          <CardContainer 
            className="card-container" 
            allState={this.state} 
          />}
        </div>
      </div>
    );
  }
}

App.propTypes = {
  cardType: PropTypes.string
};

export default App;
