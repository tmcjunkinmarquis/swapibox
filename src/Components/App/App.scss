.app {
  // width: 100%;
  // height: 90%;
  background-color: black;
}

.project-name {
  z-index: 10;
  margin-left: 40px;
  margin-bottom: 20px;
  width: 150px;
  height: 100px;
  background-color: darkorange;
  color: white;
  font-size: 25px;
  border-radius: 5px;
  
}

.buttons {
  margin-left: 30px;
  display: flex;
  flex-direction: column-reverse;
  justify-content: space-evenly;
}

.button {
  z-index: 10;
  width: 150px;
  height: 100px;
  background-color: gainsboro;
  color: rgb(43, 82, 156);
  font-size: 25px;
  border-radius: 5px;
  margin: 10px;
}

section {
  margin-left: 50px;
  width: 100%;
  height: 90%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}