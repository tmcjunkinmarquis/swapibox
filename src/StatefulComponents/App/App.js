import React, { Component } from 'react';
import FilmCrawl from '../FilmCrawl/FilmCrawl';
import './App.css';
import CardContainer from '../../StatefulComponents/CardContainer/CardContainer';

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      cardType: ''
    }
  }

  handleClick =(event)=> { 
    this.setState({cardType : event.target.value})
  }

  render() {
    return (
      <div className="App">
        <div crawl="crawl">
          <FilmCrawl />
        </div>
        <div buttons="buttons">
        <button onClick={this.handleClick} value="people" className="button">People</button>
        <button className="button">Planets</button>
        <button className="button">Vehicles</button>
        <button className="button">Favorites</button>
        </div>
        <div>
          {this.state.cardType && <CardContainer cardContainer="card-container" cardType={this.state.cardType} />}
          
        </div>
      </div>
    );
  }
}

export default App;
