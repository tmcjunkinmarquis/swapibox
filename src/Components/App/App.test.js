import React from 'react';
import App from './App.js';
import { shallow } from 'enzyme';

describe('App', () => {
    let app;
  
    beforeEach( async () => {
        app = shallow(<App />, {disableLifecycleMethods: true})
    })

    it('should match the snapshot', () => {
        expect(app).toMatchSnapshot();
    })
    
    it('should have default states', async () => {
  
       const wrapper = await shallow(<App />)
        expect(wrapper.state().film).toEqual([])
        expect(wrapper.state().people).toEqual([])
        expect(wrapper.state().vehicles).toEqual([])
        expect(wrapper.state().planets).toEqual([])
        expect(wrapper.state().cleanedCardData).toEqual([])

    })

    it('should setState', () => {

    })

    it('should call fetchApi', () => {

    })

    it('should call checkCardType with the correct params', () => {

    })
    
    it('should update vehicle ste when getVehicleData is called', () => {

    })

    it('conbineObj should return correct Obj')

})