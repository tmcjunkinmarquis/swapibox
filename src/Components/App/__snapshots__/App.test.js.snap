// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          crawl="crawl"
>
          <FilmCrawl />
</div>,
        <div
          buttons="buttons"
>
          <button
                    className="button"
                    onClick={[Function]}
                    value="people"
          >
                    People
          </button>
          <button
                    className="button"
                    onClick={[Function]}
                    value="planets"
          >
                    Planets
          </button>
          <button
                    className="button"
                    onClick={[Function]}
                    value="vehicles"
          >
                    Vehicles
          </button>
          <button
                    className="button"
          >
                    Favorites
          </button>
</div>,
        <div />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FilmCrawl />,
          "crawl": "crawl",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "buttons": "buttons",
          "children": Array [
            <button
              className="button"
              onClick={[Function]}
              value="people"
>
              People
</button>,
            <button
              className="button"
              onClick={[Function]}
              value="planets"
>
              Planets
</button>,
            <button
              className="button"
              onClick={[Function]}
              value="vehicles"
>
              Vehicles
</button>,
            <button
              className="button"
>
              Favorites
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
              "className": "button",
              "onClick": [Function],
              "value": "people",
            },
            "ref": null,
            "rendered": "People",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planets",
              "className": "button",
              "onClick": [Function],
              "value": "planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
              "className": "button",
              "onClick": [Function],
              "value": "vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorites",
              "className": "button",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            crawl="crawl"
>
            <FilmCrawl />
</div>,
          <div
            buttons="buttons"
>
            <button
                        className="button"
                        onClick={[Function]}
                        value="people"
            >
                        People
            </button>
            <button
                        className="button"
                        onClick={[Function]}
                        value="planets"
            >
                        Planets
            </button>
            <button
                        className="button"
                        onClick={[Function]}
                        value="vehicles"
            >
                        Vehicles
            </button>
            <button
                        className="button"
            >
                        Favorites
            </button>
</div>,
          <div />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FilmCrawl />,
            "crawl": "crawl",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "buttons": "buttons",
            "children": Array [
              <button
                className="button"
                onClick={[Function]}
                value="people"
>
                People
</button>,
              <button
                className="button"
                onClick={[Function]}
                value="planets"
>
                Planets
</button>,
              <button
                className="button"
                onClick={[Function]}
                value="vehicles"
>
                Vehicles
</button>,
              <button
                className="button"
>
                Favorites
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "People",
                "className": "button",
                "onClick": [Function],
                "value": "people",
              },
              "ref": null,
              "rendered": "People",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Planets",
                "className": "button",
                "onClick": [Function],
                "value": "planets",
              },
              "ref": null,
              "rendered": "Planets",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Vehicles",
                "className": "button",
                "onClick": [Function],
                "value": "vehicles",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorites",
                "className": "button",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
